sonic-xcvrd/xcvrd/xcvrd_utilities/y_cable_helper.py

```diff
@@ -22,6 +26,10 @@
 
 helper_logger = logger.Logger(SYSLOG_IDENTIFIER)
 
+y_cable_port_instances = {}
+y_cable_port_locks = {}
```

Comment:
are those locks shared between CLI and applications?

Reply:
that's correct, moving to the abstract class model, there is a lock for each port. Each api to y_cable driver will lock it for that port and only release when the API is returned. Basically all y_cable api's will be called by xcvrd only. cli will call the y_cable api through redis and invoke it per port.

Reply:
This means that presence of cli activity such show, could potentially affect our switching time, right?

Reply:
For the same port, Yes it will. Cli activities such as show hwmode, download firmware etc would slow these, but in rela production we dont anticipate to use these as such.